src/test/java/com/apuope/apuope_re/services/LogInServiceTest.java

```diff
@@ -0,0 +1,50 @@
+package com.apuope.apuope_re.services;
+
+import com.apuope.apuope_re.entities.User;
+import com.apuope.apuope_re.repositories.UserRepository;
+import com.apuope.apuope_re.dto.UserCredentials;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+class LogInServiceTest {
+    @Mock
+    private UserRepository userRepository;  // Mock the repository
+
+    @InjectMocks
+    private LogInService logInService;  // Inject mock repository into the service
```

Comment:
These comments can be deleted in my opinion

Reply:
Fixed